salt/salt-auto-update.py

```diff
@@ -64,13 +63,15 @@ def main():
         print("  job id", job_id)
         state.record_update(minion_id, job_id)
 
-
-def load_file_lines(filename):
-    try:
-        return set(x.strip() for x in open(filename))
-    except IOError:
-        return set()
-
+def get_blacklist():
+    result = subprocess.check_output(["salt", "-N", "blacklist", "--preview-target"])
+    lines = result.split(b"\n")
+    deviceSet = set()
+    for i in lines:
+        device = str(i[2:].decode("utf-8")) # each line is preceded with "- "
+        if len(device) > 0:
+            deviceSet.add(device)
+    return deviceSet
```

Comment:
Nice approach for extracting the blacklist out reliably. I was thinking you were going to have to parse the blacklist file, but this is much more robust.

Reply:
If you pass `encoding='utf8` to `check_output` the output will come back as a string instead of a bytes object. You should then be able to use `result.splitlines()` to get the lines out and avoid calling `decode` in the loop.

Reply:
As discussed we are running Python2.7 that doesn't support the `encoding` option in `check_output`